app/models/journaled/bulk_delivery.rb

```diff
@@ -41,17 +41,95 @@ def kinesis_client
     Journaled::KinesisClient.generate
   end
 
-  def requeue_failed_records!(response) # rubocop:disable Metrics/AbcSize
-    records_with_responses = records.zip(response.records)
-    errored_records = records_with_responses.select { |_record, resp| resp.error_code.present? }.map(&:first)
+  def handle_failures!(response)
+    ErrorHandler.new(response, records, app_name).handle!
+  end
+
+  class ErrorHandler
```

Comment:
Extracted a class for this cause it was getting moderately complicated.

But I didn't want to extract it to a full class cause I didn't want to add specs for it on its own, so was thinking of leaving it internal. Thoughts?

Reply:
works for me, this definitely helps keep it more organized and readable